armory/scenarios/scenario.py

```diff
@@ -355,6 +355,11 @@ def evaluate_all(self):
     def next(self):
```

Comment:
I think you need to be updating `next()` in the CarlaVideoTracking class, not the base Scenario. The former already overrides `next()` and separates the `y` from `y_patch_metadata` in the 2-tuple [here](https://github.com/twosixlabs/armory/blob/master/armory/scenarios/carla_video_tracking.py#L29).

Reply:
also the conversion from `Dict` to `List[Dict]` I'd think should be as simple as something like

```
if isinstance(y, dict)`:
    y = [y]
```

I assume there should be a cleaner way than need to reference specific keys in the base scenario class

Reply:
To your former comment, yes I can implement it within `CarlaVideoTracking` but I wasn't sure if this should be a fix for other dataloaders, I will move it to this function.

Regarding conversion from Dict the problem is that the tensors within the dictionary have an extra dimension to represent batches (batch, frames, boxes). I am iterating through that first dimension with the for loop and sending each "batch" to its own dictionary with two dimensions (frames, boxes) as each value in the resultant list. I am assuming I should still try to support multi-batch correct?

Reply:
inevitably, I think, we need a way to handle this extra batch dimension that the new tf preprocessing adds. Note with the CARLA OD PR from a while earlier, I [punted](https://github.com/twosixlabs/armory/pull/1788#issue-1470111022) on this question for now. Wherever we end up adding the logic to handle it should be as specific to the issue as possible (i.e. would rather not add to a base `scenario.py` that every Armory run touches). Ideally it would be added to the carla preprocessing code but I'm not sure how to get that to work. If we can find a way to do it there, that'd be best. Offhand, the other places I can think of would be whichever inherited `Scenario` class is most appropriate. And also in metrics code, but that changes our API as far as expected format

Reply:
Unfortunately there doesn't seem to be a way to implement this within the preprocessor code as the data is not batched when it passes through there; I have moved this fixup to `CarlaVideoTracking`, see https://github.com/jprokos26/armory/commit/4837db0e3da65322ebc3624688d9b1140222bacf.

Additionally to avoid referencing a specific key, it can be changed to:
```python
    def next(self):
        super().next()
        self.y, self.y_patch_metadata = self.y
        # Fix for tfdsv4 upgrade - separate batches into list
        if isinstance(self.y, dict):
            assert (
                len(self.y.keys()) == 1
            ), f"Expected only one key in labels dict, got {self.y.keys()}"
            key = next(iter(self.y))
            self.y = [{key: batch} for batch in self.y[key]]
        self.probe.update(y=self.y, y_patch_metadata=self.y_patch_metadata)
```
Which do you think?

Reply:
Addressed in 4837db0e3da65322ebc3624688d9b1140222bacf. @lcadalzo let me know if I should use the code snippet above instead to avoid referencing `"boxes"`.